src/main/java/com/gradle/CommonCustomUserDataDevelocityListener.java

```diff
@@ -5,14 +5,14 @@
 import com.gradle.develocity.agent.maven.api.DevelocityListener;
 import org.apache.maven.MavenExecutionException;
 import org.apache.maven.execution.MavenSession;
-import org.codehaus.plexus.component.annotations.Component;
+import org.eclipse.sisu.Description;
 
-@SuppressWarnings("unused")
-@Component(
-    role = DevelocityListener.class,
-    hint = "common-custom-user-data-develocity-listener",
-    description = "Captures common custom user data in Maven build scans"
-)
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+@Named
```

Comment:
```suggestion
@Named("common-custom-user-data-develocity-listener")
```
to follow the migration guide and keep the hint.

Reply:
Added in c6f18739bfdc991b74ede3409d111f0254d9ecfd